addons/vault/aws_secrets_manager.go

```diff
@@ -0,0 +1,95 @@
+ * limitations under the License.
+ */
+
+package vault
+
+import (
+	v1 "github.com/apache/camel-k/pkg/apis/camel/v1"
+	traitv1 "github.com/apache/camel-k/pkg/apis/camel/v1/trait"
+	"github.com/apache/camel-k/pkg/trait"
+	"github.com/apache/camel-k/pkg/util"
+	"k8s.io/utils/pointer"
+)
+
+// The Secrets Manager trait can be used to use secrets from AWS Secrets Manager
+//
+// The AWS Secrets Manager trait is disabled by default.
+//
+// A sample execution of this trait, would require
+// the following trait options:
+// -t aws-secrets-manager.enabled=true -t aws-secrets-manager.access-key="aws-access-key" -t aws-secrets-manager.secret-key="aws-secret-key" -t aws-secrets-manager.region="aws-region"
+//
+// +camel-k:trait=aws-secrets-manager.
+
+type Trait struct {
+	traitv1.Trait `property:",squash"`
+	// Enables automatic configuration of the trait.
+	Auto *bool `property:"auto" json:"auto,omitempty"`
+	// The AWS Access Key to use
+	AccessKey string `property:"access-key,omitempty"`
+	// The AWS Secret Key to use
+	SecretKey string `property:"secret-key,omitempty"`
+	// The AWS Region to use
+	Region string `property:"region,omitempty"`
+	// The adapter-specific policy to use when filling the cache (use: minimizing / maximizing). Check
+	// the component documentation if unsure
+	UseDefaultCredentialsProvider string `property:"use-default-credentials-provider,omitempty"`
+}
+
+type awsSecretsManagerTrait struct {
+	trait.BaseTrait
+	Trait `property:",squash"`
+}
+
+func NewAwsSecretsManagerTrait() trait.Trait {
+	return &awsSecretsManagerTrait{
+		BaseTrait: trait.NewBaseTrait("aws-secrets-manager", trait.TraitOrderBeforeControllerCreation),
+	}
+}
+
+func (r *awsSecretsManagerTrait) Configure(environment *trait.Environment) (bool, error) {
+	if !pointer.BoolDeref(r.Enabled, false) {
+		return false, nil
+	}
+
+	if !environment.IntegrationInPhase(v1.IntegrationPhaseInitialization) && !environment.IntegrationInRunningPhases() {
+		return false, nil
+	}
+
+	if r.UseDefaultCredentialsProvider == "" {
+		r.UseDefaultCredentialsProvider = "false"
+	}
+
+	return true, nil
+}
+
+func (r *awsSecretsManagerTrait) Apply(environment *trait.Environment) error {
+	if environment.IntegrationInPhase(v1.IntegrationPhaseInitialization) {
+		util.StringSliceUniqueAdd(&environment.Integration.Status.Capabilities, v1.CapabilityAwsSecretsManager)
+		// Add the Camel Quarkus AWS Secrets Manager
+		util.StringSliceUniqueAdd(&environment.Integration.Status.Dependencies, "mvn:org.apache.camel.quarkus:camel-quarkus-aws-secrets-manager")
+	}
+
+	if environment.IntegrationInRunningPhases() {
+		environment.ApplicationProperties["camel.vault.aws.accessKey"] = r.AccessKey
+		environment.ApplicationProperties["camel.vault.aws.secretKey"] = r.SecretKey
+		environment.ApplicationProperties["camel.vault.aws.region"] = r.Region
+		environment.ApplicationProperties["camel.vault.aws.defaultCredentialsProvider"] = r.UseDefaultCredentialsProvider
```

Comment:
From https://camel.apache.org/components/next/aws-secrets-manager-component.html#_component_options, should that be `usedefaultCredentialsProvider`

Reply:
It's defined in the vault configuration: https://github.com/apache/camel/blob/main/core/camel-main/src/generated/resources/META-INF/camel-main-configuration-metadata.json#L212

It's not consistent, we should fix this for sure. Thanks for spotting.